src/main/java/seedu/address/logic/parser/LogCommandParser.java

```diff
@@ -40,23 +48,32 @@ public LogCommand parse(String args) throws ParseException {
                             LogCommand.MESSAGE_ONE_FLAG));
                 }
                 String flag = argMultimap.getValue(PREFIX_FLAG).get();
-                assert(flag.equals(LogCommand.FLAG_RESET) || flag.equals(LogCommand.FLAG_OVERRIDE));
+                assert(flag.equals(LogCommand.FLAG_RESET) || flag.equals(LogCommand.FLAG_OVERRIDE)
+                        || flag.equals(LogCommand.FLAG_RESET_ALL));
 
                 // Reset only depends on if a name is given,
                 // It should ignore any hour provided, and should not throw an error even if hour is null.
                 if (flag.equals("r")) {
-                    return new LogCommand(studySpot, null, isNamePresent,
-                            false, !isNamePresent);
+                    return new LogCommand(studySpot, null, isNamePresent, false, !isNamePresent);
+                }
+                else if (flag.equals("ra")) {
```

Comment:
`else if` should be on the same line as the curly brace in line 58

Reply:
ok thank u!